test/testlib.h

```diff
@@ -480,6 +486,39 @@ do {                                                                           \
         zpc_ec_key_free(&ec_key);                                              \
 } while (0)
 
+# define TESTLIB_APQN_CAPS_CHECK(apqns, mkvp, type, size, flags)               \
+do {                                                                           \
+        int rc;                                                                \
+        struct zpc_aes_key *aes_key;                                           \
+        const u8 key[32] = {0};                                                \
+                                                                               \
+        rc = zpc_aes_key_alloc(&aes_key);                                      \
+        if (rc != 0)                                                           \
+            GTEST_SKIP_("APQN_CAPS check: Cannot allocate key object.");       \
+        rc = zpc_aes_key_set_type(aes_key, type);                              \
+        rc += zpc_aes_key_set_size(aes_key, size);                             \
+        rc += zpc_aes_key_set_flags(aes_key, flags);                           \
+        if (rc != 0)                                                           \
+            GTEST_SKIP_("APQN_CAPS check: Cannot set type, size, and flags."); \
+        if (apqns != NULL)                                                     \
+            rc = zpc_aes_key_set_apqns(aes_key, apqns);                        \
+        if (rc != 0)                                                           \
```

Comment:
shouldn't this if be in the same block as above `rc = zpc_aes_key_set_apqns(aes_key, apqns);` ?
Otherwise, if apqns is NULL, but rc was non-zero from previous steps, then it will skip here as well. 
Or does the GTEST_SKIP_ break the while loop, so that this can't happen?

Reply:
Yes, GTEST_SKIP_ prints the message and returns.